modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -211,8 +212,41 @@ public Object visit(Or filter, Object extraData) {
     @Override
     public Object visit(Not filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
+        // in case of a not operator we cannot simply wrap the child filter
+        // with a $not since mongo syntax is {property:{$not:{operator-expression}}}
+        // thus using a Visitor to find the PropertyName
+        class PropertyNameFinder extends FilterAttributeExtractor {
+            List<PropertyName> pNames = new ArrayList<>();
+
+            @Override
+            public Object visit(PropertyName expression, Object data) {
+                pNames.add(expression);
+                return super.visit(expression, data);
+            }
+
+            PropertyName getPropertyName() {
+                if (pNames.size() > 0) return pNames.get(0);
```

Comment:
If I understand this correctly:
1. The existing ``FilterAttributeExtractor`` can be used directly because it stores the found properties on a ``Set`` and you need to obtain the first found property?
2. This is why here you only take the first found property?

Reply:
thanks for your point @nmco , I should have clarified this with a comment before: `FilterAttributeExtractor` would extract also Literal adding them to the Set as PropertyName, so I preferred to extend it with the above method class